source/extensions/filters/http/ext_proc/matching_utils.h

```diff
@@ -0,0 +1,118 @@
+#pragma once
+
+#include "source/common/common/logger.h"
+#include "source/common/protobuf/protobuf.h"
+#include "source/extensions/filters/common/expr/evaluator.h"
+
+#if defined(USE_CEL_PARSER)
+#include "parser/parser.h"
+#endif
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace ExternalProcessing {
+
+class ExpressionManager : public Logger::Loggable<Logger::Id::ext_proc> {
+public:
+  ExpressionManager(Extensions::Filters::Common::Expr::BuilderInstanceSharedPtr builder,
+                    const Protobuf::RepeatedPtrField<std::string>& request_matchers,
+                    const Protobuf::RepeatedPtrField<std::string>& response_matchers)
+      : builder_(builder), request_expr_(initExpressions(request_matchers)),
+        response_expr_(initExpressions(response_matchers)){};
+
+  // This struct exists because the lifetime of the api expr must exceed expressions
+  // parsed from it.
+  struct ExpressionPtrWithExpr {
+    ExpressionPtrWithExpr(const google::api::expr::v1alpha1::Expr& expr,
+                          const Filters::Common::Expr::ExpressionPtr&& expr_ptr)
+        : expr_(std::move(expr)), expression_ptr_(std::move(expr_ptr)){};
+    const google::api::expr::v1alpha1::Expr& expr_;
+    const Filters::Common::Expr::ExpressionPtr& expression_ptr_;
```

Comment:
is `const ExpressionPtr&` the right type here? i think it should be just `ExpressionPtr` (no `const` or `&`) instead?

Reply:
fixed thanks!